### PR TITLE
An article about containment on root

### PR DESCRIPTION
**Pull request by mirisuzanne**
(this still needs a hero image)

## Steps to test/reproduce
_Please explain how to best reproduce the issue and/or test the changes locally (including the pages/URLs/views/states to review)._

- Mostly the new article
- I did also add some upcoming events

### CODE REVIEW

**Comment by mirisuzanne** [comment]
[I want an article on root containment](https://trello.com/c/DIysJM3e/128-i-want-an-article-on-root-containment)

File: content/blog/2023/contain-root.md [hunk 1]
```diff
@@ -0,0 +1,517 @@
...
+  will usually go un-noticed,
+  but full `size` containment
+  will collapse the box
+  unless we provide an external block-size.
+
+{{ embed.codepen(
+  id='YzRVPma',
+  title='Containment Examples',
+  user='miriamsuzanne'
+) }}
```
Comment (jgerigmeyer) [inline content/blog/2023/contain-root.md:103]: There are a few spelling errors (`Contianment`) in this codeopen.

File: content/blog/2023/contain-root.md [hunk 2]
```diff
@@ -0,0 +1,517 @@
...
+With `@container`,
+we can have each `.card` query
+the space that it lives in.
+
+But container queries provide
+a few additional features
+that are not possible with `@media`,
+and might be useful
+even when we're measuring the outermost container.
+Most importantly;
```
Comment (jgerigmeyer) [inline content/blog/2023/contain-root.md:141]:
```suggestion
Most importantly:
```

File: content/blog/2023/contain-root.md [hunk 3]
```diff
@@ -0,0 +1,517 @@
...
+  - Container Queries
+image:
+  src: blog/2023/pwc23.png
+summary: |
+  I spoke about Container Queries
+  at both Smashing Conference (San Francisco)
+  and CSS Day (Amsterdam) --
+  where I recommended setting up a root container
+  to replace most media queries.
+  Since then, Temani Afif
```
Comment (jgerigmeyer) [inline content/blog/2023/contain-root.md:18]: Link here to Temani Afif since it's the first mention?

File: content/blog/2023/contain-root.md [hunk 4]
```diff
@@ -0,0 +1,517 @@
...
+they describe the context
+immediately surrounding our web pages
+in the browser:
+the 'viewport' through which we're looking,
+the 'canvas' our site is painted onto,
+and the 'blocks' that defines
+our initial positioning and layout context.
+For simplicity,
+I'm going to refer to all of this
+collectively as The Viewport™️.
```
Comment (jgerigmeyer) [inline content/blog/2023/contain-root.md:262]: I like it personally, but curious if others might find the `™` confusing?

File: content/blog/2023/contain-root.md [hunk 5]
```diff
@@ -0,0 +1,517 @@
...
+Simple. 🥴
+
+There's a reason
+the CSS Working Group
+[has resolved](https://github.com/w3c/csswg-drafts/issues/6079#issuecomment-816307011)
+that:
+
+> RESOLVED: No future properties should propagate from `<body>` to the ICB
+
+('ICB' is the Initial Containing Block)
```
Comment (jgerigmeyer) [inline content/blog/2023/contain-root.md:372]:
```suggestion
('ICB' is the Initial Containing Block.)
```

File: content/blog/2023/contain-root.md [hunk 6]
```diff
@@ -0,0 +1,517 @@
...
+
+> RESOLVED: deprecate any existing use of body propagation
+
+Body propagation was a mistake.
+We're stuck with it now,
+but we don't have to protect it moving forward.
+
+## Containment and propagation (according to spec)
+
+[CSS Containment Module Level 2](https://drafts.csswg.org/css-contain-1/#contain-property):
```
Comment (jgerigmeyer) [inline content/blog/2023/contain-root.md:382]:
```suggestion
[CSS Containment Module Level 2](https://drafts.csswg.org/css-contain-1/#contain-property)
```

File: content/blog/2023/contain-root.md [hunk 7]
```diff
@@ -0,0 +1,517 @@
...
+but I've been trying to parse out
+how browsers got a different answer than I did.
+
+It's not clear to me
+that browsers are 'wrong'
+in the default case,
+when the root element has default `visible` overflow.
+In that case,
+I think the spec should be clarified.
+It says to propagate from body,
```
Comment (jgerigmeyer) [inline content/blog/2023/contain-root.md:438]:
```suggestion
It says to propagate from the body,
```

File: content/blog/2023/contain-root.md [hunk 8]
```diff
@@ -0,0 +1,517 @@
...
+
+{{ embed.codepen(
+  id='jOQmzzG',
+  title='Root overflow propagates fine',
+  user='miriamsuzanne'
+) }}
+
+But when we add containment,
+our non-default overflow no longer propagates.
+This is very clearly a browser bug:
```
Comment (jgerigmeyer) [inline content/blog/2023/contain-root.md:461]: Reconsider phrasing -- three uses of "clearly" or "very clearly" in quick succession.

**Review by jgerigmeyer — APPROVED** [review]

File: content/blog/2023/contain-root.md (hunk 3 above)
Comment (mirisuzanne) [reply content/blog/2023/contain-root.md:18]: I don't think we allow links in summaries? Or do we strip them out automatically in the list view?

Comment (jgerigmeyer) [reply content/blog/2023/contain-root.md:18]: I think we allow them, but strip them out for the list view. Try it and see?

File: content/blog/2023/contain-root.md [hunk 9]
```diff
@@ -0,0 +1,517 @@
...
+when we set a `background` on `html`,
+that background
+[propagates to the document canvas](https://drafts.csswg.org/css-backgrounds/#special-backgrounds)
+instead.
+If we don't set a background on `html`,
+but we do set a background on `body`,
+then the `body` background
+propagates instead --
+skipping over the root,
+and straight up to the canvas!
```
Comment (jerivas) [inline content/blog/2023/contain-root.md:286]: Wow, TIL! Surprised of the amount of little edge cases like this browsers have to support

**Review by jerivas — APPROVED** [review]

File: content/blog/2023/contain-root.md [hunk 10]
```diff
@@ -0,0 +1,540 @@
+---
+title: Can We Query the Root Container?
+sub: The complexities of containment, overflow, and 'propagation'
+author: miriam
+date: 2023-07-04
```
Comment (mirisuzanne) [inline content/blog/2023/contain-root.md:5]: I don't know if we actually want to publish tomorrow, or hold off for Wednesday?

File: content/blog/2023/contain-root.md [hunk 11]
```diff
@@ -0,0 +1,544 @@
...
+around root/body propagation and containment:
+
+> When any containments are active
+> on either the HTML `<html>` or `<body>` elements,
+> propagation of properties from the `<body>` element
+> to the initial containing block, the viewport, or the canvas background,
+> is disabled. Notably, this affects:
+>
+> - `writing-mode`, `direction`, and `text-orientation`
+>   (see CSS Writing Modes 3 § 8 The Principal Writing Mode)
```
Comment (stacyk) [inline content/blog/2023/contain-root.md:395]: @mirisuzanne I had to google this `§` sign. I don't know if others know what it means, but I didn't until today

File: content/blog/2023/contain-root.md [hunk 12]
```diff
@@ -0,0 +1,544 @@
...
+> - `overflow` and its longhands
+>   (see CSS Overflow 3 § 3.3 Overflow Viewport Propagation)
+> - `background` and its longhands
+>   (see CSS Backgrounds 3 § 2.11.2 The Canvas Background and the HTML `<body>` Element)
+
+(Maybe this is the complete list?!)
```
Comment (stacyk) [inline content/blog/2023/contain-root.md:401]: @mirisuzanne Maybe we should turn this into a call to action if someone has or knows where a list may exist, or wants to create one?

**Review by stacyk — COMMENTED** [review]
@mirisuzanne Great article. Just a few notes... also, this isn't part of the PR, but I was looking at talks, and is that turtle image https://deploy-preview-373--oddleventy.netlify.app/talks/containers/ related to the talk?

**Comment by mirisuzanne** [comment]
> this isn't part of the PR, but I was looking at talks, and is that turtle image https://deploy-preview-373--oddleventy.netlify.app/talks/containers/ related to the talk?

@stacyk - yeah, I talk about 'components that carry their container with them' using that photo of turtles.

**Review by stacyk — APPROVED** [review]

**Pull request merged**